plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -392,17 +392,18 @@ def static_env
     end
 
     def set_secrets
-      return unless needs_secret_puller?
-      set_secret_puller
+      return unless SECRET_PULLER_IMAGE
       convert_secret_env_to_annotations if secret_env_as_annotations?
+      return unless secret_annotations.any?
+      set_secret_puller
       expand_secret_annotations
     end
 
     def secret_env_as_annotations?
       ENV["SECRET_ENV_AS_ANNOTATIONS"]
     end
 
-    # storing secrets as env vars makes them visible in the deploy docs and when inspecting deployments, avoid it
+    # storing secrets as env vars makes them visible in the deploy docs and when inspecting deployments, to avoid it
     # we replace all secrtes from the env here and they are expanded by expand_secret_annotations later
```

Comment:
fix typo

Reply:
... was looking for that typo on the green line far too long :/